numpy/core/src/umath/legacy_array_method.c

```diff
@@ -273,6 +359,27 @@ PyArray_NewLegacyWrappingArrayMethod(PyUFuncObject *ufunc,
         flags = _NPY_METH_FORCE_CAST_INPUTS;
     }
 
+    get_reduction_intial_function *get_reduction_intial = NULL;
+    if (ufunc->nin == 2 && ufunc->nout == 1) {
+        npy_bool reorderable = NPY_FALSE;
+        PyObject *identity_obj = PyUFunc_GetIdentity(ufunc, &reorderable);
+        if (identity_obj == NULL) {
+            return NULL;
+        }
+        /*
+         * TODO: For object, "reorderable" is needed(?), because otherwise
+         *       we disable multi-axis reductions `arr.sum(0, 1)`. But for
+         *       `arr = array([["a", "b"], ["c", "d"]], dtype="object")`
+         *       it isn't actually reorderable (order changes result).
+         */
+        if (reorderable) {
+            flags |= NPY_METH_IS_REORDERABLE;
+        }
+        if (identity_obj != Py_None) {
+            /* NOTE: We defer, just in case it fails in weird cases: */
```

Comment:
I don't understand this comment!

Reply:
Hehe, this calls `PyArray_Pack(identity_value)`.  If you wrote a weird user dtype, maybe that actually fails for certain values (say -1).  In that case, I don't want to fail at ufunc initialization time, since maybe it never worked, but nobody would call `ufunc.reduce` later.

Reply:
OK, that makes sense!  Maybe just update the comment with what you wrote above!

Reply:
I decided to just delete it here and expand the other one (changes pending).

For NumPy, it is convenient right now to write generic code.  If we were not doing that, the functions would just use static things anway, i.e. be written as (very roughly):
```
static int
copy_initial_data<T>() {
    *<T *>initial = 0;
    return 0;
}
```
so yes, I didn't mean to make this available (at least for now).